x/oracle/keeper/prophecy.go

```diff
@@ -140,14 +142,18 @@ func (k Keeper) AppendSignature(ctx sdk.Context, prophecyID []byte, ethereumAddr
 
 // CleanUpProphecy clean up outdated prophecy
 func (k Keeper) CleanUpProphecy(ctx sdk.Context) {
+	// it is low efficient to check outdated prophecy each block
+	if k.currentHeight % CleanUpFrequency != 0 {
+		return
+	}
 	var prophecyInfo types.ProphecyInfo
 	currentHeight := uint64(k.currentHeight)
 
 	store := ctx.KVStore(k.storeKey)
 	iter := sdk.KVStorePrefixIterator(store, types.SignaturePrefix)
 	for ; iter.Valid(); iter.Next() {
 		k.cdc.MustUnmarshal(iter.Value(), &prophecyInfo)
-		if prophecyInfo.BlockNumber-currentHeight > ProphecyLiftTime {
+		if currentHeight > prophecyInfo.BlockNumber + ProphecyLiftTime {
 			storePrefix := append(types.SignaturePrefix, prophecyInfo.ProphecyId[:]...)
```

Comment:
I think we should put `append(types.SignaturePrefix, prophecyInfo.ProphecyId[:]...)` into it's own func, which takes `prophecyID` as parameter, for readability.

Reply:
Maybe in addition, put L157-158 into a `DeleteProphecyInfo` func. Same for L159-160. This makes it a lot more readable. 
It also lets us test the delete function, separate from the filter logic in L156

Reply:
Why don't we check if the prophecy status is `completed` instead of `pending`, before deleting?

Reply:
It is from requirement, if no enough relayers to finalize the prophecy, we just remove it even it is still pending.

Reply:
But in our case, the GlobalNonce is strictly increasing, ([enforced here ](https://github.com/Sifchain/sifnode/blob/future/peggy2/x/ethbridge/keeper/globalNonce.go#L39) ) so if a prophecy is removed instead of finalized, the bridge will still expect that nonce, thus the bridge halts. Am I wrong ?

Reply:
Yes, you are right. Since the lifeTime we set for prophecy is about one week. It is a very extreme condition for most of relayers have not worked for one week. so we don't check the status of prophecy when remove its signature and global nonce.

Reply:
Yes I agree this is an extremely unlikely event. If there's proposal thats unconfirmed for this long we have a much bigger problem. 
I think it is still safer to include checking if the status is completed, just because deleting a pending prophecy will require more manual fix. 
If you have bandwidth, I suggest do it now, if not, add a TODO.

Reply:
will add the status check.